components/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/handler/grant/AccountSwitchGrantHandler.java

```diff
@@ -72,6 +85,13 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         }
 
         User authorizedUser = User.getUserFromUserName(validationResponseDTO.getAuthorizedUser());
+        AccessTokenDO accessTokenDO = OAuth2Util.findAccessToken(token, false);
+        if (isSubOrganizationUser(accessTokenDO)) {
+            String authorizedUseName = resolveUserName(accessTokenDO.getAuthzUser().getAccessingOrganization(),
+                    accessTokenDO.getAuthzUser().getUserName().split("@")[0]);
```

Comment:
for emails as username, what's the impact of this?

Reply:
Fixed